tfhe/docs/fundamentals/compress.md

```diff
@@ -41,6 +41,84 @@ fn main() {
 }
 ```
 
+## Ciphertexts compressed while encrypted
+
+Ciphertexts can also be compressed while encrypted.
+They can be encrypted after any number of homomorphic operations.
+
+The API uses a list which can compress many ciphertexts of different types.
+Using a single list allows a better compression ratio.
+
+Using multiple lists (for example to compress ciphertexts at different times) is also possible but will use more space.
+
+This example shows how to compress and decompress a list containing 4 messages (one 32 bits integer, one 64 bits integer, one boolean and one 2 bits integer):
+
+```rust
+use crate::prelude::*;
+use crate::shortint::parameters::glwe_compression::COMP_PARAMS_FOR_MESSAGE_2_CARRY_2_KS_PBS;
+use crate::{
+    set_server_key, CompressedCiphertextList, CompressedCiphertextListBuilder, FheBool,
+    FheInt64, FheUint16, FheUint2, FheUint32,
+};
+
+fn main() {
+    let config = crate::ConfigBuilder::default()
+        .enable_compression(COMP_PARAMS_FOR_MESSAGE_2_CARRY_2_KS_PBS)
+        .build();
+
+    let ck = crate::ClientKey::generate(config);
+    let sk = crate::ServerKey::new(&ck);
+
+    set_server_key(sk);
+
+    let ct1 = FheUint32::encrypt(17_u32, &ck);
+
+    let ct2 = FheInt64::encrypt(-1i64, &ck);
+
+    let ct3 = FheBool::encrypt(false, &ck);
+
+    let ct4 = FheUint2::encrypt(3u8, &ck);
+
+    let compressed_list = CompressedCiphertextListBuilder::new()
+        .push(ct1)
+        .push(ct2)
+        .push(ct3)
+        .push(ct4)
+        .build()
+        .unwrap();
+
+    let serialized = bincode::serialize(&compressed_list).unwrap();
+
+    let compressed_list: CompressedCiphertextList = bincode::deserialize(&serialized).unwrap();
+
```

Comment:
here maybe print the list length if it's available

Reply:
done